### PR TITLE
FontPicker ref warnings in tests

### LINKED ISSUE #7875: FontPicker ref warnings in tests
## Task Description

Getting the following warnings when running JS unit tests:

```
Warning: Function components cannot be given refs. Attempts to access this ref will fail. Did you mean to use React.forwardRef()?

Check the render method of `FontPicker`.
    at mockConstructor (/home/runner/work/web-stories-wp/web-stories-wp/node_modules/jest-mock/build/index.js:171:19)
    at FontPicker (/home/runner/work/web-stories-wp/web-stories-wp/assets/src/edit-story/components/panels/design/textStyle/fontPicker.js:38:3)
    at div
    at row__Row (/home/runner/work/web-stories-wp/web-stories-wp/node_modules/styled-components/src/models/Keyframes.js:20:51)
    at FontControls (/home/runner/work/web-stories-wp/web-stories-wp/assets/src/edit-story/components/panels/design/textStyle/font.js:64:3)
    at div
    at content__Container (/home/runner/work/web-stories-wp/web-stories-wp/node_modules/styled-components/src/models/Keyframes.js:20:51)
    at Content (/home/runner/work/web-stories-wp/web-stories-wp/assets/src/edit-story/components/panels/panel/shared/content.js:40:20)
    at /home/runner/work/web-stories-wp/web-stories-wp/node_modules/use-context-selector/src/index.js:29:20
    at section
    at panel__Wrapper (/home/runner/work/web-stories-wp/web-stories-wp/node_modules/styled-components/src/models/Keyframes.js:20:51)
    at Panel (/home/runner/work/web-stories-wp/web-stories-wp/assets/src/edit-story/components/panels/panel/panel.js:45:3)
    at SimplePanel (/home/runner/work/web-stories-wp/web-stories-wp/assets/src/edit-story/components/panels/panel/simplePanel.js:29:24)
    at textStyle___StyledSimplePanel (/home/runner/work/web-stories-wp/web-stories-wp/node_modules/styled-components/src/models/Keyframes.js:20:51)
    at StylePanel (/home/runner/work/web-stories-wp/web-stories-wp/assets/src/edit-story/components/panels/design/textStyle/textStyle.js:34:27)
    at /home/runner/work/web-stories-wp/web-stories-wp/node_modules/use-context-selector/src/index.js:29:20
    at /home/runner/work/web-stories-wp/web-stories-wp/node_modules/use-context-selector/src/index.js:29:20
    at /home/runner/work/web-stories-wp/web-stories-wp/node_modules/use-context-selector/src/index.js:29:20
    at /home/runner/work/web-stories-wp/web-stories-wp/node_modules/use-context-selector/src/index.js:29:20
    at Wrapper (/home/runner/work/web-stories-wp/web-stories-wp/assets/src/edit-story/components/panels/design/textStyle/test/textStyle.js:58:20)
    at TestPanel (/home/runner/work/web-stories-wp/web-stories-wp/assets/src/edit-story/components/panels/shared/test/_utils.js:34:3)
    at Object.<anonymous>.exports.ThemeProvider (/home/runner/work/web-stories-wp/web-stories-wp/node_modules/styled-components/src/models/ServerStyleSheet.js:66:13)
````

Example run: https://github.com/google/web-stories-wp/pull/7855/checks?check_run_id=2772397125#step:9:12

### PR DESCRIPTION
**Pull request by spacedmonkey**
## Summary

While working on https://github.com/google/web-stories-wp/pull/8087, I fixed #7875 as it was a blocker. As a recommendation from @swissspidy I have broken this fix into it's own PR. 

### QA

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

### UAT

- [ ] UAT should use the same steps as above.

This PR can be tested by following these steps:

1.

## Checklist

- [ ] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [ ] I have tested this code to the best of my abilities
- [ ] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [ ] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [ ] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [ ] I have added documentation where necessary
- [ ] I have added a matching `Type: XYZ` label to the PR

---

Fixes #7875

### CODE REVIEW

File: assets/src/edit-story/components/panels/design/textStyle/textStyle.js [hunk 1]
```diff
@@ -30,16 +30,14 @@ import StyleControls from './style';
 import ColorControls from './color';
 import FontControls from './font';
 
-function StylePanel(props) {
-  const fontDropdownRef = useRef(null);
-  const textColorRef = useRef(null);
+const StylePanel = forwardRef(function StylePanel(props, ref) {
```
Comment (miina) [inline assets/src/edit-story/components/panels/design/textStyle/textStyle.js:33]: Doesn't look like the highlighting is working as expected here (doesn't work on `main` either, probably due to the ref issues)

@BrittanyIRL Looks like the linking/highlighting from the pre-publish checklist is not working as expected for the text panel currently (e.g. contrast) -- is this something that is being addressed within the general pre-publish checklist work already?

File: assets/src/edit-story/components/panels/design/textStyle/textStyle.js [hunk 2]
```diff
@@ -30,16 +30,14 @@ import StyleControls from './style';
...
   // use highlights to update panel styles
   // but don't dynamically adjust the `isPersistable` prop on `SimplePanel`
   // the textStyle panel automatically opens already whenever a text element is selected
   // if we update this to only be when there's a highlight the functionality that is expected
   // will be wrong.
-  const dropdownHighlight = useFocusHighlight(states.FONT, fontDropdownRef);
-  const colorHighlight = useFocusHighlight(states.TEXT_COLOR, textColorRef);
+  const dropdownHighlight = useFocusHighlight(states.FONT, ref);
+  const colorHighlight = useFocusHighlight(states.TEXT_COLOR, ref);
```
Comment (BrittanyIRL) [inline assets/src/edit-story/components/panels/design/textStyle/textStyle.js:40]: We need these separate refs to highlight the given dropdown. 

text panel highlights

File: assets/src/edit-story/components/panels/design/textStyle/textStyle.js (hunk 1 above)
Comment (BrittanyIRL) [reply assets/src/edit-story/components/panels/design/textStyle/textStyle.js:33]: So these highlights are actually for the quick action menus that appear on the side of the canvas (see the gif I attached below). 

We're ripping up the old checklist and had to rewrite all of the highlights for the checklist 😓  so we'll be retesting those thoroughly, not worried about the old checklist.

**Review by BrittanyIRL — COMMENTED** [review]
Thanks for tagging me, @miina ! 

I'd rather we not remove these separate refs here, they are what give us the quick action menu highlights of the correct inputs so it's just going to cause a new bug.  I get that we want to remove the warning. Is there another route to take to do so?

**Comment by spacedmonkey** [comment]
I dont really understand this issue and my fix doesn't work. 

I am going to close this PR out and leave others to fix this issue.

**Comment by BrittanyIRL** [comment]
> I dont really understand this issue and my fix doesn't work.
> 
> I am going to close this PR out and leave others to fix this issue.

I will tag pea pod for this issue since it's our doing with the double ref. thank you for trying to fix this!

**Pull request closed without merging**